### PR TITLE
AB#89379 Added land acquisition page

### PR DESCRIPTION
**Pull request by rafalgajewski**

### CODE REVIEW

File: src/HE.Investment.AHP.WWW/Views/Site/LandAcquisitionStatus.cshtml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+@model  SiteModel;
+
+@{
+    ViewData["Title"] = SitePageTitles.LandAcquisitionStatus;
+}
+
+<div class="govuk-grid-column-two-thirds">
+    <vc:ahp-workflow-back-button current-page="@SiteWorkflowState.LandAcquisitionStatus" />
+    @await Html.PartialAsync("~/Partials/Errors/_ErrorSummaryPartial.cshtml")
+    <vc:page-header caption="@Model.Name" />
```
Comment (Antek012) [inline src/HE.Investment.AHP.WWW/Views/Site/LandAcquisitionStatus.cshtml:17]: Use ViwBag.SiteName and remove hidden input.

File: src/HE.Investment.AHP.Contract/Site/Enums/SiteLandAcquisitionStatus.cs [hunk 2]
```diff
@@ -13,6 +15,7 @@ public enum SiteLandAcquisitionStatus
     LandGifted,
 
     [Description("Conditional acquisition, land option or heads of terms")]
+    [Hint("")]
```
Comment (Antek012) [inline src/HE.Investment.AHP.Contract/Site/Enums/SiteLandAcquisitionStatus.cs]: Why this is empty?

**Review by Antek012 — APPROVED** [review]

File: src/HE.Investment.AHP.Domain/Site/Entities/SiteEntity.cs [hunk 3]
```diff
@@ -1,5 +1,6 @@
 extern alias Org;
 
+using System.Linq.Expressions;
```
Comment (jjankowski87) [inline src/HE.Investment.AHP.Domain/Site/Entities/SiteEntity.cs:3]: not used

File: src/HE.Investment.AHP.WWW/Controllers/SiteController.cs [hunk 4]
```diff
@@ -497,6 +498,27 @@ public async Task<IActionResult> NumberOfGreenLights(SiteModel model, Cancellati
...
+        var siteModel = await GetSiteDetails(siteId, cancellationToken);
+        return View("LandAcquisitionStatus", siteModel);
+    }
+
+    [HttpPost("{siteId}/land-acquisition-status")]
+    [WorkflowState(SiteWorkflowState.LandAcquisitionStatus)]
+    public async Task<IActionResult> LandAcquisitionStatus(SiteModel model, CancellationToken cancellationToken)
+    {
+        return await ExecuteSiteCommand<SiteModel>(
+            new ProvideNumberOfGreenLightsCommand(
```
Comment (jjankowski87) [inline src/HE.Investment.AHP.WWW/Controllers/SiteController.cs:514]: wrong command and view name

File: src/HE.Investments.AHP.IntegrationTests/FillSite/Order01StartAhpSite.cs [hunk 5]
```diff
@@ -253,13 +253,24 @@ public async Task Order17_ShouldProvideNumberOfGreenLights()
...
             SitePageTitles.NumberOfGreenLights,
-            SitePagesUrl.SiteTenderingStatus(SiteData.SiteId),
+            SitePagesUrl.SiteLandAcquisitionStatus(SiteData.SiteId),
             (nameof(SiteModel.NumberOfGreenLights), "5"));
     }
 
     [Fact(Skip = AhpConfig.SkipTest)]
     [Order(18)]
-    public async Task Order18_ShouldProvideTenderingStatus()
+    public async Task Order18_ShouldProvideLandAcauisitionStatus()
```
Comment (jjankowski87) [inline src/HE.Investments.AHP.IntegrationTests/FillSite/Order01StartAhpSite.cs:262]: typo Acuisition

**Review by jjankowski87 — APPROVED** [review]

File: src/HE.Investments.AHP.IntegrationTests/FillSite/Order01StartAhpSite.cs
Comment (jjankowski87) [inline src/HE.Investments.AHP.IntegrationTests/FillSite/Order01StartAhpSite.cs:1]: merge main and fix latest test names

File: src/HE.Investment.AHP.Domain/Site/Entities/SiteEntity.cs (hunk 3 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.Domain/Site/Entities/SiteEntity.cs:3]: Removed

File: src/HE.Investments.AHP.IntegrationTests/FillSite/Order01StartAhpSite.cs (hunk 5 above)
Comment (rafalgajewski) [reply src/HE.Investments.AHP.IntegrationTests/FillSite/Order01StartAhpSite.cs:262]: fixed

File: src/HE.Investment.AHP.WWW/Controllers/SiteController.cs (hunk 4 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.WWW/Controllers/SiteController.cs:514]: fixed.

File: src/HE.Investments.AHP.IntegrationTests/FillSite/Order01StartAhpSite.cs
Comment (rafalgajewski) [reply src/HE.Investments.AHP.IntegrationTests/FillSite/Order01StartAhpSite.cs:1]: will do

**Pull request merged**